vendor/vendor-android/ddmlib/src/main/kotlin/com/malinskiy/marathon/android/AndroidAppInstaller.kt

```diff
@@ -79,9 +80,16 @@ class AndroidAppInstaller(
         val usedStorageThresholdInPercents = androidConfiguration.usedStorageThresholdInPercents
         if (storageUsedPercentage > usedStorageThresholdInPercents) {
             logger.warn { "On ${device.serialNumber} used more than $usedStorageThresholdInPercents% of storage" }
-            androidConfiguration.cleanupDeviceScript.let {
-                logger.info { "Launching cleanup shell script `$it`" }
-                device.safeExecuteShellCommand(it).let { logger.info { it } }
+            val appsToClean = device.safeExecuteShellCommand(INSTALLED_TEST_APPS_SCRIPT).split(" ")
+            logger.info { "Removing ${appsToClean.size} apps on ${device.serialNumber}" }
+            appsToClean.forEach {
+                val result = device.safeUninstallPackage(it)
+                if (result == "Success") {
+                    logger.info { "Uninstalled $it - $result" }
+                    installedApps[device.serialNumber]?.remove(it)
```

Comment:
Without it isApkInstalled check will pass and if in some other batch the tests for that apk would be launched, they will be incompleted as there will be no test apk. It can happen for a modules where we have a lot of tests.